[ko] Translate /tasks/administer-cluster/quota-api-object into Korean

Eundms opened the pull request:
<!--
 Hello!

 PLEASE title the FIRST commit appropriately, so that if you squash all
 your commits into one, the combined commit message makes sense.
 For overall help on editing and submitting pull requests, visit:
  https://kubernetes.io/docs/contribute/suggesting-improvements/

 Use the default base branch, “main”, if you're documenting existing
 features in the English localization.

 If you're working on a different localization (not English), see
 https://kubernetes.io/docs/contribute/new-content/overview/#choose-which-git-branch-to-use
 for advice.

 If you're documenting a feature that will be part of a future release, see
 https://kubernetes.io/docs/contribute/new-content/new-features/ for advice.
-->
### Description
Translate `/tasks/administer-cluster/quota-api-object` into Korean.

<!--
 Remember to ADD A DESCRIPTION and delete this note before submitting
 your pull request. The description should explain what will change,
 and why.
-->

### Issue
#44507 

<!--
 If this pull request resolves an open issue, please link the issue in the PR
 description so it will automatically close when the PR is merged.

 See the GitHub documentation for more details and other options:

 https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

Closes: #44507

developowl reviewed (commented):
Hi, Eundms. 🥝
I have reviewed this PR as Korea Contribution Academy (OSSCA).

Thank you for your hard work in working on this large amount of documentation. 

info) I also shared this on Discord, if you review the website in a local environment using a Docker container, it's easy to check if the URL is applied correctly and if there are any errors!

```
# You can set $CONTAINER_ENGINE to the name of any Docker-like container tool

# Render the full website
make container-serve

# Render only a specific language segment (e.g., English)
make container-serve segments=en

# Render multiple languages (e.g., English and Korean)
make container-serve segments=en,ko
```

Please check review!
**작업 하시느라 고생 많으셨습니다! 감사합니다.

developowl commented:
/assign

Eundms commented:
> Hi, Eundms. 🥝 I have reviewed this PR as Korea Contribution Academy (OSSCA).
> 
> Thank you for your hard work in working on this large amount of documentation.
> 
> info) I also shared this on Discord, if you review the website in a local environment using a Docker container, it's easy to check if the URL is applied correctly and if there are any errors!
> 
> ```
> # You can set $CONTAINER_ENGINE to the name of any Docker-like container tool
> 
> # Render the full website
> make container-serve
> 
> # Render only a specific language segment (e.g., English)
> make container-serve segments=en
> 
> # Render multiple languages (e.g., English and Korean)
> make container-serve segments=en,ko
> ```
> 
> Please check review! 작업 하시느라 고생 많으셨습니다! 감사합니다.

Thank you for your review.
1. I changed to use `.` instead of using `:`
2. If there is a previously translated Korean file, 
1) I changed the document path to a Korean file.
2) I changed it to follow its title.

developowl commented:
I've checked your comments. Thank you for your hard work. 🥝
This PR looks good to me!

/cc @ianychoi @jongwooo

S0okJu commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
````diff
@@ -0,0 +1,174 @@
+kubectl apply -f https://k8s.io/examples/admin/resource/quota-objects-pvc-2.yaml --namespace=quota-object-example
+```
+
+출력 결과는 네임스페이스의 쿼터를 초과하기 때문에 두 번째 PersistentVolumeClaim이 생성되지 않았음을 보여준다.
````
```suggestion
출력 결과는 네임스페이스의 쿼터를 초과하여 두 번째 퍼시스턴트 볼륨 클레임이 생성되지 않았음을 보여준다.
```

Eundms replied:
퍼시스턴트볼륨클레임(PersistentVolumeClaim)으로 표기하겠습니다

S0okJu commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
```diff
@@ -0,0 +1,174 @@
+
+## 참고
+
+쿼터로 제한할 수 있는 API 리소스를 식별하는 문자열은 다음과 같다.
```
```suggestion
다음 문자열은 쿼터로 제한할 수 있는 API 리소스를 식별하는데 사용된다.
```

Eundms replied:
다음은 쿼터로 제한할 수 있는 API 리소스를 식별하는 문자열이다. 로 하겠습니다

S0okJu commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
```diff
@@ -0,0 +1,174 @@
+## {{% heading "whatsnext" %}}
+
+
+### 클러스터 관리자용
```
~ 용 <명사> 가 나오는 것이 자연스러울 것 같아서 아래와 같이 수정해봤습니다. 
```suggestion
### 클러스터 관리자를 위한 내용
```

S0okJu commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
```diff
@@ -0,0 +1,174 @@
+
+* [네임스페이스에 대한 파드 쿼터 구성](/ko/docs/tasks/administer-cluster/manage-resources/quota-pod-namespace/)
+
+### 애플리케이션 개발자용
```
```suggestion
### 애플리케이션 개발자를 위한 내용
```

S0okJu reviewed (changes requested):
번역 수고하셨습니다! 👍

제가 드릴 수 있는 피드백은 아래와 같습니다. 
- [용어집](https://kubernetes.io/ko/docs/reference/glossary/?all=true)에 의하면 쿠버네티스 오브젝트는 외래어 표기법을 따릅니다. 
- 일부 문장은 문맥에 맞게 다듬어 봤습니다.

Eundms commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
```diff
@@ -0,0 +1,174 @@
+
+<!-- overview -->
+
+이 페이지에서는 PersistentVolumeClaims 및 Services를 포함한 API 오브젝트에 대한 쿼터를 구성하는 방법을 보여준다. 쿼터는 네임스페이스 내에서 생성할 수 있는 특정 유형 오브젝트의 개수를 제한한다. 
```
PersistentVolumeClaims는 kubectl api-resources 결과의 kind에 해당하는 API 오브젝트이므로, 외래어 표기와 영문 병기를 함께 사용하며 한글에는 대소문자 구분이 없으므로 이를 띄어쓰기 없이 붙여서 처리하는 규칙에 의해 띄어쓰기는 하지 않도록 수정하겠습니다.

Eundms commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
````diff
@@ -0,0 +1,174 @@
+kubectl create namespace quota-object-example
+```
+
+## ResourceQuota 생성
````
한글화 문서에 따라 리소스쿼터(ResourceQuota) 로 표기하겠습니다.
https://kubernetes.io/ko/docs/contribute/localization_ko/

Eundms commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
```diff
@@ -0,0 +1,174 @@
+
+## ResourceQuota 생성
+
+다음은 ResourceQuota 오브젝트에 대한 설정 파일이다.
```
한글화 문서에 따라 리소스쿼터(ResourceQuota) 로 표기하겠습니다.
https://kubernetes.io/ko/docs/contribute/localization_ko/

Eundms commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
```diff
@@ -0,0 +1,174 @@
+
+{{% code_sample file="admin/resource/quota-objects.yaml" %}}
+
+ResourceQuota를 생성한다.
```
한글화 문서에 따라 리소스쿼터(ResourceQuota) 로 표기하겠습니다.
https://kubernetes.io/ko/docs/contribute/localization_ko/

Eundms commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
````diff
@@ -0,0 +1,174 @@
+kubectl apply -f https://k8s.io/examples/admin/resource/quota-objects.yaml --namespace=quota-object-example
+```
+
+ResourceQuota의 상세 정보를 확인한다.
````
한글화 문서에 따라 리소스쿼터(ResourceQuota) 로 표기하겠습니다.
https://kubernetes.io/ko/docs/contribute/localization_ko/

Eundms commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
````diff
@@ -0,0 +1,174 @@
+pvc-quota-demo   Pending
+```
+
+## 두 번째 PersistentVolumeClaim 생성 시도
````
퍼시스턴트볼륨클레임(PersistentVolumeClaim)으로 표기하겠습니다

Eundms commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
```diff
@@ -0,0 +1,174 @@
+
+{{% code_sample file="admin/resource/quota-objects-pvc-2.yaml" %}}
+
+두 번째 PersistentVolumeClaim을 생성 시도한다.
```
퍼시스턴트볼륨클레임(PersistentVolumeClaim)으로 표기하겠습니다

Eundms commented:
> 번역 수고하셨습니다! 👍
> 
> 제가 드릴 수 있는 피드백은 아래와 같습니다.
> 
> * [용어집](https://kubernetes.io/ko/docs/reference/glossary/?all=true)에 의하면 쿠버네티스 오브젝트는 외래어 표기법을 따릅니다.
> * 일부 문장은 문맥에 맞게 다듬어 봤습니다.

리뷰 감사합니다  : )
일부 반영하였고 자세한 내용은 코멘트의 댓글을 확인부탁드립니다.

Eundms commented on content/ko/examples/admin/resource/quota-objects-pvc-2.yaml:
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
```
### 현재 상황
> 혹시 다른 방식이나 대안이 있다면 피드백 부탁드립니다.

- `quota-api-object.md` 내에서는 현재 [쿠버네티스 마크다운 문서에서 사용하는 custom hugo shortcode인 code_sample](https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#source-code-files)을 통해 예제 코드를 포함하고 있습니다. 따라서, example 파일을 포함시켰습니다. 

```markdown
<!-- quota-api-object.md 내 code_sample을 통해 포함된 파일 목록-->
{{% code_sample file="admin/resource/quota-objects-pvc.yaml" %}}
{{% code_sample file="admin/resource/quota-objects-pvc-2.yaml" %}}
{{% code_sample file="admin/resource/quota-objects.yaml" %}}
```

### `content/ko/examples`에 파일 포함시킨 이유 

-  [/layouts/shortcodes/code_sample.html](https://github.com/kubernetes/website/blob/main/layouts/shortcodes/code_sample.html) 을 확인해봤을 때, 한국어 페이지를 렌더링할 때는 `.Page.Lang`이 `ko`로 고정되기 때문에 fail이 되는 것 같습니다. 

```go
{{ $filename := printf "/content/%s/examples/%s" .Page.Lang $file | safeURL }}

{{ $resource := readFile $filename}}

{{ with $resource }}{{ $.Scratch.Set "content" . }}

{{ if not ($.Scratch.Get "content") }}
  {{ errorf "[%s] %q not found in %q" site.Language.Lang $fileDir.File $bundlePath }}
{{ end }}
```
  - 빌드 에러 확인한 방법
      - 로컬 : `make module-init` -> `make container-serve segments=en,ko` 
      - 풀 리퀘스트 오픈 후, netlify 가 진행하는 검사에서 fail 
  - 참고한 파일/문서 : , [custom hugo shortcode 생성 방법](https://gohugo.io/content-management/shortcodes/#create-your-own-shortcodes) 


### 추후  
- 말씀주신대로 불필요하게 관리 공수가 늘어나는 문제를 피하기 위해, `code_sample.html`에 한글화 대상이 아닌 예제 파일은 en 디렉터리에서 참조하는 fallback 로직을 추가하여, 빌드가 가능하도록 수정하는 방식을 고려할 수 있을 것 같습니다.

Eundms replied:
@ianychoi Sure, I will create a new issue to track the implementation of the fallback logic.

Eundms commented:
> @Eundms Thank you for your contribution (and for the review activities from the Korea Contribution Academy (OSSCA) mentees).
> 
> Overall, the content looks good to me. I have a few additional comments on this PR.



> @Eundms Thank you for your contribution (and for the review activities from the Korea Contribution Academy (OSSCA) mentees).
> 
> Overall, the content looks good to me. I have a few additional comments on this PR.


Thank you for the feedback 🙂
I have committed the changes reflecting all comments except for the `examples files`.
If you have any other suggestions or alternative approaches, please let me know!

hyorimlee commented:
Hi, @Eundms.
I have reviewed this PR as Korea Contribution Academy - Challengers program. This commit looks good to me.
/cc @jongwooo @ianychoi

hyorimlee commented:
/assign

ianychoi reviewed (commented):
/assign
/lgtm

seokho-son commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
```diff
@@ -0,0 +1,177 @@
+API 오브젝트에 대한 쿼터를 구성하는 방법을 보여준다. 쿼터는 네임스페이스 내에서 
+생성할 수 있는 특정 유형의 오브젝트 개수를 제한한다. 
+쿼터는 
+[ResourceQuota](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#resourcequota-v1-core)
```
FYI: https://kubernetes.io/ko/docs/contribute/localization_ko/#한글화-용어집


English | 한글 |비고
-- | -- | --
ResourceQuota | 리소스쿼터(ResourceQuota) | API 오브젝트인 경우


```suggestion
[리소스쿼터(ResourceQuota)](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#resourcequota-v1-core)
```

seokho-son commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
```diff
@@ -0,0 +1,177 @@
+생성할 수 있는 특정 유형의 오브젝트 개수를 제한한다. 
+쿼터는 
+[ResourceQuota](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#resourcequota-v1-core)
+오브젝트에서 지정한다.
```
오브젝트를 통해서 지정한다는 의미로써

```suggestion
오브젝트로 지정한다.
```

가 좀 더 자연스러운 것 같습니다. 
(꼭 모든 리뷰 의견에 따를 필요는 없습니다. ^^)

seokho-son commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
````diff
@@ -0,0 +1,177 @@
+kubectl get resourcequota object-quota-demo --namespace=quota-object-example --output=yaml
+```
+
+출력 결과는 quota-object-example 네임스페이스에서 퍼시스턴트볼륨클레임은 최대 1개, LoadBalancer 타입 서비스는 최대 2개, 그리고 NodePort 타입 서비스는
+허용되지 않음을 보여준다.
````
```suggestion
출력 결과를 통해 quota-object-example 네임스페이스에서 퍼시스턴트볼륨클레임은
최대 1개, LoadBalancer 타입 서비스는 최대 2개가 허용되며, NodePort 타입 서비스는
허용되지 않음을 확인할 수 있다.
```

조금 더 자연스럽게 수정해보았습니다.
또한, 원문과 개행을 맞추기도 다시 확인 부탁 드립니다.

<img width="480" height="44" alt="Image" src="https://github.com/user-attachments/assets/12ee2836-e58f-42c1-a647-571d0675ea1d" />

seokho-son commented on content/ko/docs/tasks/administer-cluster/quota-api-object.md:
````diff
@@ -0,0 +1,177 @@
+kubectl apply -f https://k8s.io/examples/admin/resource/quota-objects-pvc-2.yaml --namespace=quota-object-example
+```
+
+출력 결과는 네임스페이스의 쿼터를 초과하기 때문에 두 번째 퍼시스턴트볼륨클레임이 생성되지 않았음을 보여준다.
````
```suggestion
출력 결과는 네임스페이스의 쿼터 초과에 의해서 두 번째
퍼시스턴트볼륨클레임이 생성되지 않았음을 보여준다.
```

<img width="407" height="26" alt="Image" src="https://github.com/user-attachments/assets/8e5c4a0f-8fa8-4197-a75c-2e6412ccf5c6" />

seokho-son reviewed (commented):
@Eundms 리뷰 의견 반영 감사합니다. :)

추가적으로 보완하면 좋은 사항들이 있어서, 리뷰 의견 남겼습니다.
